doc/source/async_api.rst

```diff
@@ -1,78 +1,94 @@
-+----------------------------------------+-----------------------------------------------------+
-| .. code-block:: python                 | .. code-block:: python                              |
-|                                        |                                                     |
-|   # Execute f asynchronously.          |   # Execute f asynchronously with Ray/asyncio.      |
-|                                        |                                                     |
-|                                        |   from ray.experimental import async_api            |
-|                                        |                                                     |
-|                                        |   @ray.remote                                       |
-|   async def f():                       |   def f():                                          |
-|       await asyncio.sleep(1)           |       time.sleep(1)                                 |
-|       return 1                         |       return 1                                      |
-|                                        |                                                     |
-|                                        |   ray.init()                                        |
-|   loop = asyncio.get_event_loop()      |   loop = asyncio.get_event_loop()                   |
-|   tasks = [f() for i in range(4)]      |   tasks = [async_api.as_future(f.remote())          |
-|                                        |            for i in range(4)]                       |
-|   results = loop.run_until_complete(   |   results = loop.run_until_complete(                |
-|       asyncio.gather(tasks))           |       asyncio.gather(tasks))                        |
-+----------------------------------------+-----------------------------------------------------+
+    import asyncio
+
+    @ray.remote
+    class AsyncActor:
+        async def run_task(self):
+            print("started")
+            await asyncio.sleep(1) # Network, I/O task here
+            print("ended")
+
+    actor = AsyncActor.remote()
+    # Expect all 50 tasks started at once, after 1 seconds
+    # they should finish at the same time
+    ray.get([actor.run_task.remote() for _ in range(50)])
+
+Under the hood, Ray runs all the method inside a single python event loop.
+Please note that running blocking ``ray.get`` or ``ray.wait`` inside async
+actor method is not allowed. ``ray.get`` will block the execution of the event
+loop.
+
+You can limit the number of concurrent task running at once using the
+``max_concurrency`` flag. By default, only 1000 tasks can be running at once.
+
+.. code-block:: python
+
+    import asyncio
+
+    @ray.remote
+    class AsyncActor:
+        async def run_task(self):
+            print("started")
+            await asyncio.sleep(1) # Network, I/O task here
+            print("ended")
+
+    actor = AsyncActor.options(max_concurreny=10).remote()
```

Comment:
Can we also set this without using `.options()`? If not we should probably add it.